### PR TITLE
Implemented error extraction from logs files on server side using grep

### LINKED ISSUE #632: LogAnalyzer can not handle large files
Check https://github.com/SatelliteQE/satellite6-upgrade/issues/83

### PR DESCRIPTION
**Pull request by renzon**
close #632

### CODE REVIEW

**Comment by renzon** [comment]
@sghai @jyejare @san7ket: instead of bringing all diff on logs during upgrade to local python and analysing it locally I've used grep to analyse errors on server side and bringing only errors through network. This way I expect less overhead on it. The downside is not having the entire log content.

But once I asked you guys about the need of those entire log files and it seems you have another ways of accessing them, it would not be an issue. Anyway I've also added some logs to check the time of code execution so at least I have more clues in case this fix doesn't work on the job itself.

File: automation_tools/satellite6/log.py [hunk 1]
```diff
@@ -123,38 +137,15 @@ def _save_full_log(host, log_file_path, content):
...
     """Just a wrapper to make mocking easier on tests"""
-    print(s.encode('utf-8'))
+    if isinstance(s, unicode):
+        print(s.encode('utf-8'))
+    else:
+        print(s)
+
+
+if __name__ == '__main__':
+    with LogAnalyzer('root@cloud-qe-15.idmqe.lab.eng.bos.redhat.com',
```
Comment (san7ket) [inline automation_tools/satellite6/log.py]: Hard-coded Value ?

Comment (renzon) [reply automation_tools/satellite6/log.py]: I've forgot to remove the code a used to test against a real host. Problem fixed.

**Comment by san7ket** [comment]
There was a failure recently on the job due loganalyzer. when there was a non utf8 character in the log file

```
Traceback (most recent call last):
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/fabric/main.py", line 762, in main
    *args, **kwargs
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/fabric/tasks.py", line 425, in execute
    results['<local-only>'] = task.run(*args, **new_kwargs)
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/fabric/tasks.py", line 172, in run
    return self.wrapped(*args, **kwargs)
  File "/home/jenkins/workspace/upgrade-to-6.3-rhel7/upgrade/runner.py", line 233, in product_upgrade
    host=cap_host)
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/automation_tools/satellite6/log.py", line 80, in __exit__
    execute(fetch_appended_log_lines, host=self.host)
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/fabric/tasks.py", line 385, in execute
    multiprocessing
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/fabric/tasks.py", line 275, in _execute
    return task.run(*args, **kwargs)
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/fabric/tasks.py", line 172, in run
    return self.wrapped(*args, **kwargs)
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/automation_tools/satellite6/log.py", line 78, in fetch_appended_log_lines
    analyze(self.host, log_file, content)
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib/python2.7/site-packages/automation_tools/satellite6/log.py", line 146, in analyze
    content_lines = content.decode('utf-8').split('\n')
  File "/home/jenkins/shiningpanda/jobs/47b1311d/virtualenvs/d41d8cd9/lib64/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe0 in position 813131: invalid continuation byte
```

**Comment by renzon** [comment]
Please don't merge yet, I am going to fix https://github.com/SatelliteQE/automation-tools/issues/632#issuecomment-342875037 on this branch

**Pull request merged**